pkg/health/health.go

```diff
@@ -0,0 +1,158 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/metrics"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/pkg/errors"
+)
+
+// Diagnosis is the information about the health of the revision.
+type Diagnosis struct {
+	EnoughRequests     bool
+	CheckResults       []*CheckResult
+	FailedCheckResults []*CheckResult
+	IsHealthy          bool
```

Comment:
ideally this structure should also represent that no rollout operation should be done (halt/noop) based on minrequests.

similar to IsHealthy but different.

this is probably an enum DiagnosisResult { UNKNOWN=0 , INCONCLUSIVE, HEALTHY= , UNHEALTHY }